models/report.js

```diff
@@ -129,4 +147,65 @@ Report.findSortedPage = function(filter, page, callback) {
   });
 };
 
+Report.checkoutBatch = function(userId, callback) {
+  async.series([ 
+    this.releaseBatch, 
+    this.lockBatch.bind(this, userId, ITEMS_PER_BATCH),
+    this.loadBatch.bind(this, userId, ITEMS_PER_BATCH)
+  ], function(err, results) {
+    if (err) return callback(err);
+    callback(null, results[2]);
+  });
+}
+
+// release items in older batches
+Report.releaseBatch = function(callback) {
+  var conditions = { checkedOutAt: { $lt: timeAgo(BATCH_TIMEOUT) } };
+  var update = { checkedOutBy: null, checkedOutAt: null };
+
+  Report.update(conditions, update, { multi: true }, callback);
+}
+
+// cancel batch for given user
+Report.cancelBatch = function(userId, callback) {
+  var conditions = { checkedOutBy: userId };
+  var update = { checkedOutBy: null, checkedOutAt: null };
+  
+  Report.update(conditions, update, { multi: true }, callback);
+}
+
+// lock a new batch for given user
+Report.lockBatch = function(userId, limit, callback) {
+  var conditions = {
+    checkedOutAt: null, 
+    checkedOutBy: null,
+    read: false
+  };
+
+  Report.find(conditions).sort({storedAt: -1}).limit(limit).exec(function(err, reports) {
+    if (err) return callback(err);
+    var ids = reports.map(function(report) { return report._id; });
+    var update = { checkedOutBy: userId, checkedOutAt: new Date() };
+    Report.update({ _id: { $in: ids } }, update, { multi: true }, callback);
+  });
+}
+
+// load a batch for user
+Report.loadBatch = function(userId, limit, callback) {
+  var conditions = {
+    checkedOutAt: { $ne: null }, 
+    checkedOutBy: userId
+  };
+
+  Report.find(conditions).limit(limit).exec(callback);
+}
+
```

Comment:
With all these functions called ___Batch it's feeling like the batch functionality should be refactored into a separate class. No need for a new model, just a POJO. What do you think?